server/src/Room.ts

```diff
@@ -636,12 +678,22 @@ export class Room extends EnhancedEventEmitter<RoomEvents> {
 		broadcasterPeer.on('new-producer', ({ producer }) => {
 			const peers = this.getAllPeers();
 
-			for (const peer of peers) {
-				void peer.consume({
-					producer,
-					consumerReplicas: this.#consumerReplicas,
+			this.producerRouter
+				.pipeToRouter({
+					producerId: producer.id,
+					router: this.consumerRouter,
+				})
+				.then(() => {
+					for (const peer of peers) {
```

Comment:
NOTE: I'm redoing this PR making pipe transports optional. But just a comment about this code:

It's wrong because here you are collecting peers before `await pipeToRouter()` and hence, when `pipeToRouter()` resolves peers may have changed.